XenAdmin/TabPages/PhysicalStoragePage.cs

```diff
@@ -328,74 +282,59 @@ private void newSRButton_Click(object sender, EventArgs e)
             new NewSRCommand(Program.MainWindow, connection).Execute();
         }
 
-        /// <summary>
-        /// Taken from http://support.microsoft.com/kb/319401.
-        /// </summary>
-        /// <param name="sender"></param>
-        /// <param name="e"></param>
-        private void listViewSrs_ColumnClick(object sender, ColumnClickEventArgs e)
+        private void dataGridViewSr_MouseUp(object sender, MouseEventArgs e)
         {
-            // Determine if clicked column is already the column that is being sorted.
-            if (e.Column == lvwColumnSorter.SortColumn)

```

Comment:
The logici n this method hasa not been replicated in the datagridview and currently sorting is broken. For example, sorting by ascending Size should give `0B, 1024MB, 423.7GB, 3686.4GB`, but currently `3686.4GB` comes before `423.7GB`.

Reply:
This is a very valid point. Thanks for your comment!